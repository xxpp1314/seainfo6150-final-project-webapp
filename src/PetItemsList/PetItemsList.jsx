import React from "react";
import PropTypes from "prop-types";
import PetItem from "../PetItem/PetItem.jsx";
import styles from "./PetItemsList.module.css";
import { Link } from "react-router-dom";

const PetItemsList = props => {

  return (
    <div className={styles.flexContainer}>
      <h1 className={styles.title}>{props.pagetitle}</h1>
      <ul className={styles.container}>
            {props.petItemList.map((petItem,index) => (
                <li key={index}>
                  <Link  key={index} className={styles.link} to={'/'+petItem.category+'/'+petItem.categoryid}>
                    <PetItem petItem={petItem}/>
                  </Link>
                </li>
            ))}
      </ul>
    </div>
  );
};

PetItemsList.propTypes = {
  petItemList: PropTypes.array.isRequired,
  pagetitle: PropTypes.string.isRequired
};

export default PetItemsList;