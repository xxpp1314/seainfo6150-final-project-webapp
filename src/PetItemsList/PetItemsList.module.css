.container{
    padding:2% 2% 2% 2%;
    background-image: linear-gradient(to right,rgb(83, 15, 15), rgb(49, 12, 25) 70%);
    gap:1%;
    list-style: none;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    font-family: 'Lora', serif;
    font-size: medium;
    font-weight: lighter;
    height: 100%;
  }
  
  .flexContainer{
    display: flex;
    flex-direction: column;
    text-align: center;
  }
  
  .title{
    padding-top: 1%;
    background-image: linear-gradient(to right,rgb(83, 15, 15), rgb(49, 12, 25) 70%);
    color: rgb(221, 193, 193);
    font-family: 'Lora', serif;
    box-shadow: 0px 0px 5px 2px rgba(204, 0, 0, 0.5);    
    border: 0px solid grey;
  }
  
  .link{
    text-decoration-line: none;
  }
  
  @media screen and (min-width: 180px) {
    .container {
      grid-template-columns: repeat(1, 1fr);
      padding-bottom: 70%;
    }
  }
  
  @media screen and (min-width: 200px) {
    .container {
      grid-template-columns: repeat(1, 1fr);
      padding-bottom: 50%;
    }
  }
  
  @media screen and (min-width: 500px) {
    .container {
      grid-template-columns: repeat(2, 1fr);
      padding-bottom: 5%;
    }
  }
  
  @media screen and (min-width: 900px) {
    .container {
      grid-template-columns: repeat(3, 1fr);
      padding-bottom: 2%;
    }
  }