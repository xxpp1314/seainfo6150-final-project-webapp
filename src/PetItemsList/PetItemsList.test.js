import React from "react";
import { render } from "@testing-library/react";
import PetItemsList from "./PetItemsList.jsx";
import { BrowserRouter } from 'react-router-dom';

const petItem={
    "Russell":
    {
       "price":"$300",
       "title":"Russell",
       "category":"dogs",
       "categoryid":"Russell",
       "img":
       {
          "url":"https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/50829956/1/?bust=1619138292&width=1080",
          "height":400,
          "width":500
       }
    }
}

it("renders the PetItemsList correctly", () => {
  const { container } = render(
    <BrowserRouter>
        <PetItemsList petItemList={Object.values(petItem)} pagetitle="Dogs"/>
    </BrowserRouter>
    );
  expect(container).toMatchSnapshot();
});
