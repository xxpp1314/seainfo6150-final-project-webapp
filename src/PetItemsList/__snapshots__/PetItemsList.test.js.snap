// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PetItemsList correctly 1`] = `
<div>
  <div
    class="flexContainer"
  >
    <h1
      class="title"
    >
      Dogs
    </h1>
    <ul
      class="container"
    >
      <li>
        <a
          class="link"
          href="/dogs/Russell"
        >
          <div
            class="petItem"
          >
            <p
              class="text"
            >
              Russell
            </p>
            <div>
              <img
                alt=""
                class="image"
                src="https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/50829956/1/?bust=1619138292&width=1080"
              />
            </div>
            <p
              class="text"
            >
              Price: 
              $300
            </p>
          </div>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
